web/src/components/typography/text.tsx

```diff
@@ -1,30 +1,45 @@
+import clsx from 'clsx'
 import React from 'react'
 import { useTheme } from 'react-jss'
 import { TextColorType } from './helper'
 
 import { useStyles } from './styles'
 
-export type TextProps = {
+export type Props = {
   size: 'body-large' | 'body-base' | 'body-small'
-  children: string | JSX.Element
+  children: string | JSX.Element | Element[] | Array<JSX.Element>
   emphasis?: boolean
   italic?: boolean
   strike?: boolean
   color?: TextColorType
+  className?: string
 }
 
+export type DefaultTextProps = {
+  as?: 'div' | 'span'
+}
+
+export type LabelProps = {
+  as: 'label'
+  htmlFor?: string
+}
+
+export type TextProps = Props & (DefaultTextProps | LabelProps)
+
 export const Text: React.FC<TextProps> = props => {
   const theme = useTheme()
-  const { children, size } = props
+  const { children, as = 'div', className } = props
 
   const classes = useStyles({
     ...props,
     theme,
   })
 
-  const Element = size === 'body-small' ? 'label' : 'div'
+  const Component = as
```

Comment:
Oh, cool that we're not using label anymore. Because the usage of it created some problems for "Tabs" component